ReactWindows/ReactNative.Net46/Modules/Storage/AsyncStorageModule.cs

```diff
@@ -15,6 +16,8 @@ namespace ReactNative.Modules.Storage
     class AsyncStorageModule : NativeModuleBase
     {
         private readonly SemaphoreSlim _mutex = new SemaphoreSlim(1, 1);
+        private CancellationTokenSource _cancelSource = new CancellationTokenSource();
```

Comment:
Since we don't cancel anywhere except in OnReactInstanceDispose(), it is highly unlikely that we will ever find this object in cancelled state. Most of the time it will be in disposed state.
So why pick a cancellation source?
Instead why not just maintain a bool (isDisposed?) (whose reads and writes are guaranteed to be atomic by .Net) and just set it to true in OnReactInstanceDispose().
That would reduce the complexity of the fix. Remove the need to throw/catch exceptions.
The mutex is probably unavoidable.